File: branches/Medusa/Source/Fantasy.Jobs/Resources/TaskRuntimeScheduleResourceProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.Jobs.Management;

namespace Fantasy.Jobs.Resources
{
    public class TaskRuntimeScheduleResourceProvider : RuntimeScheduleResourceProviderBase
    {
        protected override bool InternalCanHandle(string name)
        {
            return string.Equals("RunTask", name, StringComparison.OrdinalIgnoreCase); 
        }

        public override void Initialize()
        {
            base.Initialize();
            JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsPropertyChanged);
        }


        void SettingsPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
        {
            if (e.PropertyName == "TaskRuntimeSchedule")
            {
                this.Reschedule();
            }
        }

        protected override RuntimeScheduleSetting GetSetting(string id)
        {

            RuntimeScheduleSetting rs = JobManagerSettings.Default.TaskRuntimeSchedule.Tasks.Find(setting => string.Equals(String.Format("{0}|{1}", setting.Namespace, setting.Name), id, StringComparison.OrdinalIgnoreCase));
            if (rs == null)
            {
                rs = JobManagerSettings.Default.TaskRuntimeSchedule;
            }
            return rs;
        }

        protected override string GetResourceId(ResourceParameter parameter)
        {
            return string.Format("{0}|{1}", parameter.Values["namespace"], parameter.Values["taskname"]);
        }
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/resources/TaskRuntimeScheduleResourceProvider.java
package fantasy.jobs.resources;

import Fantasy.Jobs.Management.*;

public class TaskRuntimeScheduleResourceProvider extends RuntimeScheduleResourceProviderBase
{
	@Override
	protected boolean InternalCanHandle(String name)
	{
		return String.equals("RunTask", name, StringComparison.OrdinalIgnoreCase);
	}

	@Override
	public void Initialize()
	{
		super.Initialize();
//C# TO JAVA CONVERTER TODO TASK: Java has no equivalent to C#-style event wireups:
		JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsPropertyChanged);
	}


	private void SettingsPropertyChanged(Object sender, System.ComponentModel.PropertyChangedEventArgs e)
	{
		if (e.PropertyName.equals("TaskRuntimeSchedule"))
		{
			this.Reschedule();
		}
	}

	@Override
	protected RuntimeScheduleSetting GetSetting(String id)
	{

//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		RuntimeScheduleSetting rs = JobManagerSettings.getDefault().getTaskRuntimeSchedule().getTasks().Find(setting => String.equals(String.format("%1$s|%2$s", setting.Namespace, setting.getName()), id, StringComparison.OrdinalIgnoreCase));
		if (rs == null)
		{
			rs = JobManagerSettings.getDefault().getTaskRuntimeSchedule();
		}
		return rs;
	}

	@Override
	protected String GetResourceId(ResourceParameter parameter)
	{
		return String.format("%1$s|%2$s", parameter.getValues().get("namespace"), parameter.getValues().get("taskname"));
	}
}